client/app/components/home/home.controller.js

```diff
@@ -1,22 +1,32 @@
 class HomeController {
-  constructor($state, $document, $mdDialog, GroupService) {
+  constructor($state, $document, $mdDialog, GroupService, Authentication) {
     "ngInject";
     Object.assign(this, {
       $state,
       $document,
       $mdDialog,
-      GroupService
+      GroupService,
+      Authentication
     });
   }
 
   $onInit() {
-    this.GroupService.listMy().then((data) => {
-      if (data.length > 0) {
-        this.$state.go("group", { groupId: data[0].id });
-      } else {
-        this.openJoinGroupDialog();
-      }
-    });
+    if (angular.isUndefined(this.Authentication.data)) {
+      this.Authentication.update();
```

Comment:
This is good learning though I'll admit that the hookProvider coding seems a bit complex to wrap my head around.